crates/sui-core/src/authority.rs

```diff
@@ -1021,6 +1020,13 @@ impl AuthorityState {
         );
         let signer = transaction.sender();
         let gas = transaction.gas();
+        let move_vm = Arc::new(
+            adapter::new_move_vm(
+                self._native_functions.clone(),
```

Comment:
Probably a topic for a future PR, but I am unsure why `native_functions` is a field of `self` rather than constructed on-the-fly--it seems like every usage of that field is a clone.

Reply:
I agree and I did not understood it either. I looked at the code of the function that creates the natives table and it appears to be involving a bit, so I thought someone did the math and figured cloning was much better.
I initially removed it in my PR for exactly the reason you mentioned and then I figured I would leave it there...